extensions/amp-a4a/amp-a4a-format.md

```diff
@@ -0,0 +1,472 @@
+
+#### CSS Animations and Transitions
+
+##### Selectors
+
+The `transition` and `animation` properties are only allowed on selectors that:
+- Contain only `transition`, `animation`, `transform`, `visibility`, or 
+  `opacity` properties.
+- Start with `.amp-animate` followed by a space.
+
+  _Rationale:_ This allows the AMP runtime to remove this  class from context
+   to deactivate animations, when necessary for page performance.
+
+**Good**
+```css
+.amp-animate .box {
+    transform: rotate(180deg);
+    transition: transform 2s;
+}
+```
+
+**Bad**
+
+Property not allowed in CSS class.
+```css
+.amp-animate .box {
+    color: red;  // non-animation property not allowed in animation selector
+    transform: rotate(180deg);
+    transition: transform 2s;
+}
+```
+
+Missing context class `.amp-animate`.
+```css
+.box {
+    transform: rotate(180deg);
+    transition: transform 2s;
+}
+```
+
+##### Transitionable and animatable properties
+
+The only properties that may be transitioned are opacity and transform.
+([Rationale](http://www.html5rocks.com/en/tutorials/speed/high-performance-animations/))
+
+**Good**
+```css
+transition: transform 2s;
+```
+
+**Bad**
+```css
+transition: background-color 2s;
+```
+
+**Good**
+```css
+@keyframes turn {
+  from {
+    transform: rotate(180deg);
+  }
+  
+  to {
+    transform: rotate(90deg);
+  }
+}
+```
+
+**Bad**
+```css
+@keyframes slidein {
+  from {
+    margin-left:100%;
+    width:300%
+  }
+  
+  to {
+    margin-left:0%;
+    width:100%;
+  }
+}
+```
+
+
+### AMP Extensions and Builtins
+
+The following are _allowed_ AMP extension modules and AMP builtin tags in an 
+A4A creative. Extensions or builtin tags not explicitly allowed are prohibited.
+
+Most of the omissions are either for performance or to make A4A creatives 
+simpler to analyze.
+
+_Example:_ `<amp-ad>` is omitted from this list.  It is explicitly disallowed
+because allowing an `<amp-ad>` inside an `<amp-ad>` could potentially lead to
+unbounded waterfalls of ad loading, which does not meet A4A performance goals.
+
+_Example:_ `<amp-iframe>` is omitted from this list.  It is disallowed 
+because ads could use it to execute arbitrary Javascript and load arbitrary 
+content. Ads wanting to use such capabilities should return `false` from 
+their
+[a4aRegistry](https://github.com/ampproject/amphtml/blob/master/ads/_a4a-config.js#L40)
+entry and use the existing '3p iframe' ad rendering mechanism.
+
+_Example:_ `<amp-ad-network-*-impl>` are omitted from this list.  The 
+`<amp-ad>` tag handles delegation to these implementation tags; creatives 
+should not attempt to include them directly.
+
+<table>
+  <tr><td>amp-accordion</td></tr>
+  <tr><td>amp-analytics</td></tr>
+  <tr><td>amp-anim</td></tr>
+  <tr><td>amp-audio</td></tr>
+  <tr><td>amp-brid-player</td></tr>
+  <tr><td>amp-brightcove</td></tr>
+  <tr><td>amp-carousel</td></tr>
+  <tr><td>amp-dailymotion</td></tr>
+  <tr><td>amp-experiment</td></tr>
```

Comment:
In the sense, "wouldn't do anything useful"?  Or in the sense, "would break user experience or the AMP runtime"?  This comes back to the policy issue that I raised above.